lib/srf.js

```diff
@@ -871,15 +871,34 @@ class Srf extends Emitter {
         res.send(500);
       }
       else {
-        req.on('cancel', () => {
+        req.on('cancel', (cancelReq) => {
           debug('createB2BUA: received CANCEL from A party, sending CANCEL to B');
           res.send(487) ;
-          uacReq.cancel() ;
+          uacReq.cancel({
+            headers: copyUASHeaderToUAC(cancelReq)
```

Comment:
what if there is no cancelReq?  This doesnt seem backwards compatible

Reply:
if there is no cancelReq, function copyUASHeaderToUAC return empty object {}, nothing will be added for UAC cancel. so it's still backwards compatible